### PR TITLE
Search results

### PR DESCRIPTION
**Pull request by adammeyer**

### CODE REVIEW

**Comment by adammeyer** [comment]
Sorry for the many commits, I'll squash those.

File: src/app/searchResults/searchResults.component.js [hunk 1]
```diff
@@ -12,29 +12,52 @@ let componentName = 'searchResults';
...
   }
 
   $onInit(){
-    let searchTerm = urlQueryTerm.get('q');
-    if(searchTerm){
-      this.designationsService.createSearch(searchTerm)
-        .subscribe((id) => {
-          this.searchId = id;
-          this.getPageResults(1);
+    var params = this.$location.search();
```
Comment (OzzieOrca) [inline src/app/searchResults/searchResults.component.js]: Isn't this going to fail without the router?

Comment (adammeyer) [inline src/app/searchResults/searchResults.component.js]: @Omicron7 found a solution, he committed it directly to master. https://github.com/CruGlobal/give-web/commit/2302727087cdb011a0d6f1cdf308099f81d1dc35

File: src/common/services/api/designations.service.spec.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  });
+
+  describe('productSearch', () => {
+    it('should send a request to API and get results', () => {
+      self.$httpBackend.expectGET('https://cortex-gateway-stage.cru.org/search?keyword=steve').respond(200, searchResponse);
+      self.designationsService.productSearch({
+        keyword: 'steve'
+      })
+        .subscribe((data) => {
+          expect(data.length).toEqual(searchResponse.response.docs.length);
```
Comment (OzzieOrca) [inline src/common/services/api/designations.service.spec.js]: I think you should check to make sure at least the first one is mapped correctly and has the keys you expect.

File: src/app/searchResults/searchResults.component.js [hunk 3]
```diff
@@ -12,29 +12,52 @@ let componentName = 'searchResults';
...
+
+    this.searchParams.keyword = params.q;
+    this.searchParams.first_name = params.fName;
+    this.searchParams.last_name = params.lName;
```
Comment (OzzieOrca) [inline src/app/searchResults/searchResults.component.js]: What's the purpose of copying these if `requestSearch()` copies them again? Do they need to be there even if the type isn't `people`?

File: src/app/searchResults/searchResults.component.js [hunk 4]
```diff
@@ -12,29 +12,52 @@ let componentName = 'searchResults';
...
+    this.searchParams.type = params.type;
+    this.searchParamsInitial = angular.copy(this.searchParams);
+
+    this.requestSearch();
+  }
+
+  requestSearch(type){
+    if(angular.isDefined(type)){
```
Comment (OzzieOrca) [inline src/app/searchResults/searchResults.component.js]: Is `requestSearch()` called by anything other than `$onInit` when `this.searchParams` is set? Do you think this first block of logic would make more sense with what you currently have in `$onInit`? I'm a little confused why it seems like you are copying the first and last name into initial and then back.

**Comment by OzzieOrca** [comment]
There are some translate tags missing but mostly looks good.

File: src/app/searchResults/searchResults.component.js (hunk 1 above)
Comment (OzzieOrca) [inline src/app/searchResults/searchResults.component.js]: Ah ok cool

File: src/app/searchResults/searchResults.component.js (hunk 4 above)
Comment (adammeyer) [inline src/app/searchResults/searchResults.component.js]: So if they choose ministries, it doesn't show the first and last name in the fields.  I thought that would be confusing to the user.

Comment (adammeyer) [inline src/app/searchResults/searchResults.component.js]: Yes, requestSearch can be called from the view.

File: src/app/searchResults/searchResults.component.js (hunk 1 above)
Comment (Omicron7) [inline src/app/searchResults/searchResults.component.js]: We didn't have $locationProvider configured properly on production. Wasn't an issue with routers at all. You still need to use $window.location to reload or change the url, but we can use $location to read/write query params now.

Comment (OzzieOrca) [inline src/app/searchResults/searchResults.component.js]: Nice

**Pull request merged**